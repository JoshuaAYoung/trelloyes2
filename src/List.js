import React from 'react';
import Card from './Card'
import './List.css';

class List extends React.Component {
  render() {
    return (
        <section className='List'>
            <header className='List-head'>
            <h2>{this.props.header}</h2>
            </header>
            
        <div className='List-Cards'>
            {this.props.cards.map((card) =>
            <Card
              title={card.title}
              content={card.content}
              id={card.id}
              key={card.id}
              deleteCard={this.props.deleteCard}
            />
          )}
          <button
            type='button'
            className='List-add-button'
            onClick={() => this.props.randomCard()}
          >
            + Add Random Card
          </button>
        </div>
      </section>
    )
  }
}

  export default List;