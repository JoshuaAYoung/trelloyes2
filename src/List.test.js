import React from 'react';
import ReactDOM from 'react-dom';
import List from './List.js';
import renderer from 'react-test-renderer';
import { exportAllDeclaration } from '@babel/types';


describe('List component', () => {
  it('renders without crashing', () => {

    const div = document.createElement('div');

    ReactDOM.render(<List header="testHeader" 
    cards={[]}
    key="a"/>,  div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('Snapshot test', () => {
  const tree = renderer.create(<List header = "testHeader"
  cards={[]}
  key="a"/>).toJSON();
  expect(tree).toMatchSnapshot(); 
});










