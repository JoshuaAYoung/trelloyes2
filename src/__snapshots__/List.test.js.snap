// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<section
  className="List"
>
  <header
    className="List-head"
  >
    <h2>
      testHeader
    </h2>
  </header>
  <div
    className="List-Cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the list as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-head"
  >
    <h2>
      testHeader
    </h2>
  </header>
  <div
    className="List-Cards"
  >
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
